aaelf64-morello/aaelf64-morello.rst

```diff
@@ -0,0 +1,831 @@
+    +-------+-------------------------+-----------------------------------------+------------------------------------------+
+    | 59396 | ``R_MORELLO_IRELATIVE`` | ``CAP_INIT(S, A, CAP_SIZE, CAP_PERM)``  | See note below                           |
+    |       |                         |                                         |                                          |
+    +-------+-------------------------+-----------------------------------------+------------------------------------------+
+    | 59397 | ``R_MORELLO_TLSDESC``   | ``TLSDESC(S+A)``                        | Identifies a TLS descriptor to be filled |
+    |       |                         |                                         |                                          |
+    +-------+-------------------------+-----------------------------------------+------------------------------------------+
+
+.. note::
+
+  ``R_MORELLO_CAPINIT`` instructs the runtime or dynamic loader to create a 16-byte
+  capability at ``r_offset``. ``r_offset`` must be 16-byte aligned. An object
+  producer may communicate a hint about the size of the capability to the static
+  linker in the 16-byte fragment identified by ``r_offset``. The fragment has
+  the following format:
+
+  ``| 64-bits empty | 64-bits size |``
+
+  ``R_MORELLO_GLOB_DAT`` instructs the runtime or dynamic loader to create a 16-byte
+  capability in the GOT entry identified by ``r_offset``. The capability holds the
+  address of a data symbol which must be resolved at load time when dynamic
+  linking.
+
+  ``R_MORELLO_JUMP_SLOT`` instructs the dynamic loader to create a 16-byte capability
+  in the GOT entry identified by r_offset. The capability holds the address of a
+  function symbol which must be resolved at load time.
+
+  ``R_MORELLO_RELATIVE`` represents an optimization of ``R_MORELLO_GLOB_DAT``. It can be
+  used when the symbol resolves to the current shared object or executable. ``S``
+  must be the ``Null`` symbol (Index 0) the address and permissions must be written
+  to the fragment. See `Dynamic linking with Morello`_ for details.
+
+  ``R_MORELLO_IRELATIVE`` is used by the linker when transforming ``IFUNC`` s. The
+  rest are the same as ``R_MORELLO_RELATIVE``
+
+
+Static linking with Morello
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A capability has more associated information than a conventional pointer. It has
+extra information, for example base, offset, size and permissions.
+
+Capabilities cannot be statically initialised. Global capability initialization
+when static linking is performed by the runtime at program startup. The
+communication between the static linker and runtime is implementation defined.
+This document describes an implementation implemented based on a table of
+capability descriptions created at static link time, where each capability
+generating relocation results in one entry in the table. When static linking,
```

Comment:
generating -> generates

Reply:
capability-generating is an adjective. The subject that generates the table entry is the capability-generating relocation. So changed it to "where each capability-generating relocation results in one entry in the table"